nestjs/src/opportunities/opportunities.controller.ts

```diff
@@ -49,6 +54,27 @@ export class OpportunitiesController {
     return new ResumeDto(resume, students, gratitudes, feedbacks);
   }
 
+  @Patch('/:githubId/resume')
+  @ApiOperation({ operationId: 'saveResume' })
+  @ApiForbiddenResponse()
+  @ApiBadRequestResponse()
+  @ApiNotFoundResponse()
+  @ApiOkResponse({ type: Resume })
+  @UseGuards(DefaultGuard)
+  // TODO: deal with 500 status code validation issue on assigning dto FormDataDto type + @Body() decorator
```

Comment:
you dto has constructor, if i recall correctly nest won't pass your data into it

Reply:
I tried to use dto without constructor, it has different errors
![whitelist-allowed-error](https://user-images.githubusercontent.com/34455330/196049201-c900b784-7e82-4480-9b7c-3653a6ad07a9.JPG)
![whitelist-error](https://user-images.githubusercontent.com/34455330/196049204-44aa437a-5954-4db5-bcd7-e3bf392a00d0.JPG)

Reply:
@AlreadyBored you need to define validation for your fields. You just defined OpenAPI decorators

Reply:
actually, in your case I think the problem is constructor. You need to split request and response dtos.

Reply:
I removed constructor and it still doesn't work, please take a look at latest version :(

Reply:
I tried to remove constructor, it still doesn't work, please take a look at the latest version. We have a lot of similar dtos without validation decorators and it works well Oo

Reply:
Resolved